Gem/Code/Source/Components/UI/HUDComponent.cpp

```diff
@@ -25,6 +25,8 @@ namespace MultiplayerSample
     void HUDComponent::Deactivate()
     {
         m_waitForActiveNetworkMatchComponent.RemoveFromQueue();
+        m_roundNumberHandler.Disconnect();
```

Comment:
nit: where do these attach? Assume its because this is some type of base component?

Reply:
It connects once ActiveNetworkMatchComponent. UI lives on a different entity then the network stuff, so we cant use the normal component dependency to activate UI once the network match component is activated. 
An alternative could be to broadcast a notification when the NetworkMatchComponent turns on, and then listen for that message in UI.
See where m_waitForActiveNetworkMatchComponent is defined in the header.